src/amo/components/CollectionAddAddon/index.js

```diff
@@ -65,15 +73,34 @@ export class CollectionAddAddonBase extends React.Component<InternalProps> {
   };
 
   componentWillReceiveProps(props: InternalProps) {
-    const { hasAddonBeenAdded: hasAddonBeenAddedNew } = props;
-    const { hasAddonBeenAdded } = this.props;
-    if (hasAddonBeenAdded !== hasAddonBeenAddedNew) {
+    const {
+      hasAddonBeenAdded: hasAddonBeenAddedNew,
+      hasAddonBeenRemoved: hasAddonBeenRemovedNew,
+    } = props;
+    const { errorHandler, hasAddonBeenAdded, hasAddonBeenRemoved } = this.props;
+
+    const addStatusChanged = hasAddonBeenAdded !== hasAddonBeenAddedNew;
+    const removeStatusChanged = hasAddonBeenRemoved !== hasAddonBeenRemovedNew;
+
+    if (addStatusChanged) {
+      this.props.setUIState({
+        completedAddonAction: hasAddonBeenAddedNew ? addonAddedAction : null,
+      });
+    }
+
+    if (removeStatusChanged) {
       this.props.setUIState({
-        addonWasAdded: props.hasAddonBeenAdded,
+        completedAddonAction: hasAddonBeenRemovedNew
+          ? addonRemovedAction
+          : null,
       });
     }
 
-    if (hasAddonBeenAddedNew && hasAddonBeenAddedNew !== hasAddonBeenAdded) {
+    if (
+      (addStatusChanged || removeStatusChanged) &&
+      (hasAddonBeenAddedNew || hasAddonBeenRemovedNew)
+    ) {
+      errorHandler.clear();
```

Comment:
Usually this happens automatically because the saga which is using the error handler will clear it the next time you run the saga. I see why it's necessary here. Hey, it works 🤷‍♀️